kufar/books/main.go

```diff
@@ -68,13 +96,29 @@ type Database struct {
 
 var location, _ = time.LoadLocation("Europe/Minsk")
 
+// returns true is the session is new and vice-versa
+func checkSession(w http.ResponseWriter, r *http.Request) bool {
```

Comment:
You mixed up some stuff in your function.
If you function called `checkSession`, why it making some redirects and writes error to writer?)

I suggest to remove all redirects from here and leave only checks :)